syntax = "proto3";

package firmament;

message ObtainObjectMessage {
    bytes object_id = 1;
    bytes data = 2;
    uint64 size = 3;
    string sender_uuid = 4;
    string sender_uri = 5; // Check whether the two  actually needed
}


message StorageDiscoverMessage {
    string uuid = 1;
    string uri = 2;
    string storage_uri = 3 ;
}

message StorageHeartBeatMessage {
    string uri = 1 ;
    string uuid = 2; /* Is this redundant? */
    repeated ObjectInfoMessage object = 3 ;
}

message ObjectInfoMessage {
    bytes obj_name = 1;
/* Should locations be resource ids or direct host:port (storage engine
listening interface)? */
    string location = 2;
}


